src/include/migraphx/find_concur.hpp

```diff
@@ -0,0 +1,243 @@
+#ifndef MIGRAPHX_GUARD_FIND_CONCUR_HPP
+#define MIGRAPHX_GUARD_FIND_CONCUR_HPP
+
+#include <cassert>
+#include <string>
+#include <functional>
+#include <memory>
+#include <type_traits>
+#include <utility>
+#include <unordered_map>
+#include <vector>
+
+#include <migraphx/instruction.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+struct program;
+
+#ifdef DOXYGEN
+
+/// An interface for target-dependent analysis to find concurrent instructions
+/// executing in different streams.
+struct find_concur
+{
+    void get_concur(program* p,
+                    int num_of_streams,
```

Comment:
The number of stream should be passed during construction not when calling.

Reply:
in tools/include,  do not find any example of having a data field in struct.